crates/shared/src/event_handling.rs

```diff
@@ -161,24 +263,162 @@ where
         if !have_deleted_old_events {
             self.store.replace_events(Vec::new(), range.clone()).await?;
         }
-        self.last_handled_block = Some(range.end().to_u64());
+
+        let blocks = self
+            .block_retriever
+            .blocks(RangeInclusive::new(
+                range.end().saturating_sub(MAX_REORG_BLOCK_COUNT),
+                *range.end(),
+            ))
+            .await?;
+
+        self.update_last_handled_blocks(&blocks);
```

Comment:
Why do we need to do this for old_blocks? I think this will always be overriden by latest_blocks. Because if there are old_blocks then latest_blocks already contains the last 50 blocks and old_blocks the rest. Then in `update_events_from_latest_blocks` we call `update_last_handled_blocks` again which only keeps the last 25 blocks undoing the ones we added here.

Reply:
Because of the potential failure of the `update_events_from_latest_blocks` call.

Reply:
Resolving the comment but feel free to unresolve if additional comments are needed.